File: projects/ExpenseTracker/ExpenseTracker/Models/Expense.swift
//
//  Expense.swift
//  ExpenseTracker
//
//  Created by Fathi Ali on 2025-06-22.
//

import Foundation

struct Expense: Identifiable, Codable {
    let id: UUID
    var title: String
    var amount: Double
    var category: String
    var date: Date

    init(id: UUID = UUID(), title: String, amount: Double, category: String, date: Date = Date()) {
        self.id = id
        self.title = title
        self.amount = amount
        self.category = category
        self.date = date
    }
}

File: learning/java/section/Expense.java
import java.math.BigDecimal;
import java.time.LocalDate;

public class Expense {
    private final LocalDate date;
    private final String category;
    private final BigDecimal amount;
    private final String note;

    public Expense(LocalDate date, String category, BigDecimal amount, String note) {
        this.date = date;
        this.category = category;
        this.amount = amount;
        this.note = note;
    }
}